site.ts

```diff
@@ -220,22 +221,38 @@ export function inject({ config, posthog }) {
         })
     }
 
-    posthog?.getActiveMatchingSurveys((surveys) => {
-        surveys.forEach((survey) => {
-            const surveyName = survey.name.replace(/\s/g , "-")
-            if (!localStorage.getItem(`seenSurvey_${surveyName}_${survey.id}`)) {
-                const shadow = createShadow(style(surveyName, survey?.appearance))
-                const surveyPopup = createSurveyPopup(survey, surveyName)
-                addListeners(surveyPopup, surveyName, survey.id, survey.name)
-                shadow.appendChild(surveyPopup)
-                window.dispatchEvent(new Event('PHSurveyShown'))
-                const sessionRecordingUrl = getSessionRecordingUrl()
-                posthog.capture('survey shown', {
-                    $survey_name: survey.name,
-                    $survey_id: survey.id,
-                    sessionRecordingUrl: sessionRecordingUrl,
-                })
+    const callSurveys = (posthog) => {
+        posthog?.getActiveMatchingSurveys((surveys) => {
+            surveys.forEach((survey) => {
+                if (document.querySelectorAll("div[class^='PostHogSurvey']").length === 0) {
+                    const surveyName = survey.name.replace(/\s/g, "-")
+                    if (!localStorage.getItem(`seenSurvey_${surveyName}_${survey.id}`)) {
```

Comment:
Reminder to remove surveyName from here as soon as possible, since breaking change.

I think if we introduce a force refresh mechanism, it shouldn't be via the name.